src/Elastic.Apm/Model/Transaction.cs

```diff
@@ -697,7 +697,28 @@ public void End()
 		}
 
 		if (IsSampled || _apmServerInfo?.Version < new ElasticVersion(8, 0, 0, string.Empty))
+		{
```

Comment:
Implementation is fine and follows established coding style but we should probably split this long method into smaller methods, e.g this whole block.

Not important to do now just a general note.